### PR TITLE
Add support for other unix base system

### PR DESCRIPTION
**Pull request by Ernold11**

### CODE REVIEW

File: moler/config/connections.py [hunk 1]
```diff
@@ -50,7 +50,7 @@ def set_defaults():
 
 def register_builtin_connections(connection_factory, moler_conn_class):
     _register_builtin_connections(connection_factory, moler_conn_class)
-    if platform.system() == 'Linux':
+    if platform.system() in ['Linux', "FreeBSD", "Darwin", "SunOS"]:
         _register_builtin_unix_connections(connection_factory, moler_conn_class)
```
Comment (marcin-usielski) [inline moler/config/connections.py]: else and warn (Ususpported system {}".format(platform.system()) ?

File: moler/device/textualdevice.py [hunk 2]
```diff
@@ -35,7 +35,8 @@ class TextualDevice(object):
     not_connected = "NOT_CONNECTED"
     connection_hops = "CONNECTION_HOPS"
 
-    def __init__(self, sm_params=None, name=None, io_connection=None, io_type=None, variant=None, initial_state=None):
+    def __init__(self, sm_params=None, name=None, io_connection=None, io_type=None, variant=None,
+                 io_constructor_kwargs={}, initial_state=None):
```
Comment (marcin-usielski) [inline moler/device/textualdevice.py]: New argument as last one?

Comment (Ernold11) [reply moler/device/textualdevice.py]: I think is better when all io_connection parameters are together.

File: moler/config/connections.py (hunk 1 above)
Comment (Ernold11) [reply moler/config/connections.py]: done raise MolerException on unsupported systems

**Pull request merged**